Bluetooth: Audio: Upstream Coordinated Set Identification Service

Thalley opened the pull request:
Upstreams the implementation and test of the coordinated set identification service (CSIS) server and client. 

The code is still somewhat a work in progress, and no public API has been added for this yet, but we decided to start upstreaming the LE Audio implementations, even if they are not fully complete.

Thalley committed:
Bluetooth: Audio: Add coordinated sets identification service server

Adds the coordinated set identification service (CSIS) server.

This is still a work in progress and thus there are no public
API for it yet, and some code changes will still be needed.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Add coordinated sets identification service client

    Adds the coordinated set identification service (CSIS) client.

    This is still a work in progress and thus there are no public
    API for it yet, and some code changes will still be needed.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Add CSIS shell

Adds the CSIS shell implementation, to interactively control
the CSIS server.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Add CSIS client shell

Add support for the CSIS client shell to interactively
use the CSIS client.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Add CSIS BSIM tests

Add a couple of babblesim tests for the CSIS server
and client.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

asbjornsabo reviewed (approved):
The integration seems OK.
The content of the main files I have reviewed before, so assuming there are no changes, they should also be OK.

Thalley committed:
Bluetooth: Audio: Move csis.h to include/bluetooth/audio

Move the header file csis.h from the internal location to
the public include directory. This file is supposed to provide
the public API for CSIS and the CSIS client, but is not fully
complete yet.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

cfriedt commented:
@Thalley - please rebase / fix merge conflicts.
